cloud/docker.go

```diff
@@ -110,10 +110,16 @@ func (m *dockerManager) SpawnHost(ctx context.Context, h *host.Host) (*host.Host
 		return nil, errors.Wrapf(err, "Invalid Docker settings in distro '%s'", h.Distro.Id)
 	}
 
+	parent, err := distro.FindOne(distro.ById(h.ParentID))
+	if err != nil {
+		return nil, errors.Wrapf(err, "Error finding parent of host %s", h.Id)
+	}
+
+	hostIP := "host_ip"
```

Comment:
I'd have `hostIP := (*parent.ProviderSettings)["host_ip"]` unless you expect that to change over the course of the function (it seems weird that it might).

additionally, I know why provider settings is a pointer, but it might make sense to write a method just for additional safety. (or just have a method which wraps the mapstructure.Decode, or similar)

Reply:
Wrote a method on the distro to check for a nil pointer and then return the HostIP.